### PR TITLE
Allow the docker tag for the sliding sync proxy to be specified in env

### PR DESCRIPTION
**Pull request by t3chguy**
---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Review by kegsay — APPROVED** [review]

**Comment by kegsay** [comment]
CI seems to be failing though, is it able to pull the image?

File: cypress.config.ts [hunk 1]
```diff
@@ -30,6 +30,9 @@ export default defineConfig({
         experimentalSessionAndOrigin: true,
         specPattern: 'cypress/e2e/**/*.{js,jsx,ts,tsx}',
     },
+    env: {
+        SLIDING_SYNC_PROXY_TAG: "v0.4.0",
+    },
```
Comment (richvdh) [inline cypress.config.ts:35]: shouldn't this have a comment? what does it do?

File: cypress/plugins/sliding-sync/index.ts [hunk 2]
```diff
@@ -114,8 +114,10 @@ async function proxyStop(instance: ProxyInstance): Promise<void> {
  * @type {Cypress.PluginConfig}
  */
 export function slidingSyncProxyDocker(on: PluginEvents, config: PluginConfigOptions) {
+    const dockerTag = config.env["SLIDING_SYNC_PROXY_TAG"];
```
Comment (richvdh) [inline cypress/plugins/sliding-sync/index.ts:117]: this feels like it needs documentation, too.

If it were me, I'd move the comment at line 25 ("// A cypress plugins [sic] to ...") to the top of the file, and add notes on the env vars it uses under that.

**Review by richvdh — CHANGES_REQUESTED** [review]
I feel like this deserves more comments/documentation.

File: cypress/plugins/sliding-sync/index.ts [hunk 3]
```diff
@@ -22,7 +22,8 @@ import { dockerExec, dockerIp, dockerRun, dockerStop } from "../docker";
 import { getFreePort } from "../utils/port";
 import { SynapseInstance } from "../synapsedocker";
 
-// A cypress plugins to add command to start & stop https://github.com/matrix-org/sliding-sync
+// A cypress plugin to add command to start & stop https://github.com/matrix-org/sliding-sync
+// SLIDING_SYNC_PROXY_TAG env used as the docker tag to use for `ghcr.io/matrix-org/sliding-sync-proxy` image.
```
Comment (richvdh) [inline cypress/plugins/sliding-sync/index.ts:26]: shouldn't this be at the top of the file? It's a bit lost here.

**Review by richvdh — APPROVED** [review]
seems good otherwise

File: cypress/plugins/sliding-sync/index.ts (hunk 3 above)
Comment (t3chguy) [reply cypress/plugins/sliding-sync/index.ts:26]: If you put it above the copyright then it'll fail lint

Comment (richvdh) [reply cypress/plugins/sliding-sync/index.ts:26]: I humbly submit that both the linter and your IDE are braindead.

**Pull request merged**